code/topology/OOTB_install.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+OCFILE="../../blogs/scripted-windows/oc-creds.yaml"
+YAMLFILE="../../blogs/scripted-windows/"
+
+# Check if the bak file exists and exit the script
+#in order to make sure that nothing is overwritten
+
+if [ -f "$OCFILE.bak" ]; then
+	echo "config file exists in $OCFILE.bak"
+        exit 1
+fi
+
+
+# Sets the connection to the OpenShift cluster
+echo "Please enter OpenShift login details:"
+
+echo "OpenShift API:"
+read OCAPI
+
+echo "OpenShift Username:"
+read OCUSER
+
+echo "OpenShift Pass:"
+read OCPASS
+
+# Encode the values to fit the yaml 
+API=$(echo "$OCAPI" | base64)
+USER=$(echo "$OCUSER" | base64)
+PASS=$(echo "$OCPASS" | base64)
+
+sed -i.bak "s/<base64 URL>/$API/" $OCFILE
+sed -i.bak "s/<base64 user>/$USER/" $OCFILE
```

Comment:
Multiple sed commands with the backup option will not preserve the original file, you will only get the file state after the second to last sed. If the goal is to make this idempotent, combine the sed substitutions into one block, and create a new file (no in-place).

Then you can change the test for existence of the .bak file to a test for existence of the populated credentials file, and instead of exiting, use that to skip all the questions and sed.